hudi-utilities/src/main/java/org/apache/hudi/utilities/schema/SchemaRegistryProvider.java

```diff
@@ -81,11 +66,22 @@ private static Schema getSchema(String registryUrl) throws IOException {
 
   @Override
   public Schema getSourceSchema() {
-    return schema;
+    String registryUrl = config.getString(Config.SRC_SCHEMA_REGISTRY_URL_PROP);
+    try {
+      return getSchema(registryUrl);
+    } catch (IOException ioe) {
+      throw new HoodieIOException("Error reading source schema from registry :" + registryUrl, ioe);
+    }
   }
 
   @Override
   public Schema getTargetSchema() {
-    return targetSchema;
+    String registryUrl = config.getString(Config.SRC_SCHEMA_REGISTRY_URL_PROP);
+    String targetRegistryUrl = config.getString(Config.TARGET_SCHEMA_REGISTRY_URL_PROP, registryUrl);
+    try {
+      return getSchema(targetRegistryUrl);
```

Comment:
it might result in target schema != source schema when targetRegistryUrl is not specified as schema might change between getSourceSchema, getTargetSchema calls. Is it a problem?

Reply:
Should not be a problem I believe since sourceSchema is used for reading while targetSchema is used for writing. Any issues if at all should be handled in HoodieAvroUtils when we try to rewrite the record with targetSchema.

Reply:
yes, target schema is allowed to be different than source schema due to transformations and this is fine.